frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +538,164 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto isDoublePlus = std::get<bool>(info);
+  if (isDoublePlus) {
+    wr.heading(kind_, type_, loc, "'++' is not a pre-increment.");
+  } else {
+    wr.heading(kind_, type_, loc, "'--' is not a pre-decrement.");
+  }
+}
+
+void ErrorNewWithoutArgs::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto expr = std::get<const uast::AstNode*>(info);
+  wr.heading(kind_, type_, loc,
+             "'new' expression is missing its argument list.");
+  wr.message("Perhaps you intended to write 'new ", expr, "()' instead?");
+}
+
+void ErrorUseImportNeedsModule::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto isImport = std::get<bool>(info);
+  std::string useOrImport = isImport ? "import" : "use";
+  wr.heading(kind_, type_, loc, "'", useOrImport,
+             "' statements must refer to module", (isImport ? "" : " or enum"),
+             " symbols.");
+}
+
+void ErrorExceptOnlyInvalidExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto limitationKind = std::get<uast::VisibilityClause::LimitationKind>(info);
+  wr.heading(kind_, type_, loc, "incorrect expression in '", limitationKind,
+             "' list, identifier expected.");
+}
+
+void ErrorLabelIneligibleStmt::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto maybeStmt = std::get<const uast::AstNode*>(info);
+  if (!maybeStmt) {
+    wr.heading(kind_, type_, loc, "cannot label statement.");
+  } else if (maybeStmt->tag() == uast::AstTag::EmptyStmt) {
+    // this is the case where there is a semicolon following the label name
+    wr.heading(kind_, type_, loc,
+               "labels should not be terminated with semicolons.");
+  } else {
+    wr.heading(kind_, type_, loc, "cannot label '",
+               tagToString(maybeStmt->tag()), "' statement.");
+  }
+  wr.message("Only 'for', 'while', and 'do-while' statements can have labels.");
+}
+
+void ErrorBisonMemoryExhausted::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "memory exhausted while parsing.");
+}
+
+void ErrorBisonUnknownError::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<1>(info);
+  auto nearestToken = std::get<2>(info);
+  wr.heading(kind_, type_, loc,
+             "unknown error from Bison parser: ", errorMessage, ".");
+  if (!nearestToken.empty()) {
+    wr.note(loc, "Nearest token when error was encountered: '", nearestToken,
+            "'");
+  }
+}
+
+void ErrorBisonSyntaxError::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto nearestToken = std::get<std::string>(info);
+  wr.heading(kind_, type_, loc, "syntax error",
+             (nearestToken.empty() ? "" : " near '" + nearestToken + "'"), ".");
+}
```

Comment:
To preserve backwards compatibility, let's not print the words "syntax error" here, since they are already printed by the message heading (see the .good files with 'near' in them, like test/parallel/taskPar/vass/taskintents-badintents.bad.

Reply:
I see your point, but I'm also worried about if we remove "syntax error" from here, the first `syntax error:` will display with nothing following the colon which will look awkward. Is there something else you can think of to put here?

Reply:
Well, I think that's the status quo (the words "syntax error" definitely don't appear in the error message output of the production compiler prior to this). You could have `(no location given)` instead of the empty string if the nearest token is empty.